lib/ansible/modules/crypto/openssl_csr.py

```diff
@@ -254,16 +302,79 @@ def generate(self, module):
         if module.set_fs_attributes_if_different(file_args, False):
             self.changed = True
 
-    def remove(self):
-        '''Remove the Certificate Signing Request.'''
+    def check(self, module, perms_required=True):
+        """Ensure the resource is in its desired state."""
+        state_and_perms = super(CertificateSigningRequest, self).check(module, perms_required)
 
-        try:
-            os.remove(self.path)
-        except OSError as exc:
-            if exc.errno != errno.ENOENT:
-                raise CertificateSigningRequestError(exc)
+        self.privatekey = crypto_utils.load_privatekey(self.privatekey_path, self.privatekey_passphrase)
```

Comment:
No it is also needed when signing the certificate request (see L289) and i think it is ugly to open the private key twice. Opening it in __init__ is of course a very bad idea so this is actually the only place where i can be sure that the key gets opened before an existing CSR gets checked or a new one is signed.